src/Components/OpenInExploreLogsButton.tsx

```diff
@@ -0,0 +1,63 @@
+import { useReturnToPrevious } from '@grafana/runtime';
+import { LinkButton } from '@grafana/ui';
+import React, { useMemo } from 'react';
+
+export interface OpenInExploreLogsButtonProps {
+  datasourceUid?: string;
+  labelMatchers: Array<{ name: string; value: string }>;
+  from?: string;
+  to?: string;
+  returnToPreviousSource?: string;
+  renderButton?: (props: { href: string }) => React.ReactElement<any>;
+}
+
+export default function OpenInExploreLogsButton({
+  datasourceUid,
+  labelMatchers,
+  from,
+  to,
+  returnToPreviousSource,
+  renderButton,
+}: OpenInExploreLogsButtonProps) {
+  const setReturnToPrevious = useReturnToPrevious();
+
+  const href = useMemo(() => {
+    const mainLabel = labelMatchers[0];
+
+    if (!mainLabel) {
+      return null;
+    }
+
+    const url = new URL(
+      `${window.location.origin}/a/grafana-lokiexplore-app/explore/${mainLabel.name}/${mainLabel.value}/logs`
```

Comment:
```suggestion
      `${window.location.origin}/a/grafana-lokiexplore-app/explore/${mainLabel.name}/${replaceSlash(mainLabel.value)}/logs`
```

Reply:
Otherwise a label value with a slash, like loki/distributor will break the routing

Reply:
This will fail if Grafana runs with a suburl. I think 
```suggestion
      `${config.appSubUrl}/a/grafana-lokiexplore-app/explore/${mainLabel.name}/replaceSlash(${mainLabel.value})/logs`
```
with `config` from `@grafana/runtime` should do the trick, though.